tooling/github-configuration.md

```diff
@@ -0,0 +1,103 @@
+| Write | Specific individuals or teams with the "Write" permission can make contributions to the repository. Grant to active InnerSource participants with relevant skills. |
+| Admin | Limited to a select group responsible for managing the repository and overseeing InnerSource initiatives. Handles repository configuration and review processes. |
+
+It is better to choose No Policy and let the user choose, but it is also a good idea to consciously choose Read permission.
+
+#### Repository creation
+
+Allowing most users to create only Private or Internal type repositories is a measure often taken.
+It's important to consider your organization's policies and the nature of the projects when deciding on repository creation settings. Striking a balance between encouraging collaboration and managing access to sensitive information is crucial in an InnerSource context. Providing options for members to create repositories, while defining guidelines and permissions, can support innovation and knowledge sharing within the organization.
+
+| Repository Creation | Description | Recommendation |
+| --- | --- | --- |
+| No Policy | Organization owners can choose whether to allow members to create repositories. | Provide flexibility for members to create repositories based on project needs, encouraging collaboration. |
+| Disabled | Members will not be able to create repositories. | Discourages repository creation and may hinder InnerSource practices. Consider enabling creation for collaboration. |
+| Members can create repositories | Members are allowed to create repositories, but with restrictions on the types available (public, private, internal). | Encourage the creation of repositories while specifying the types that align with your organization's guidelines. |
+| ┣ Public | Members can create public repositories visible to everyone on GitHub. | Promote open sharing and collaboration with both internal and external stakeholders. |
+| ┣  Private | Members can create private repositories accessible only to invited collaborators. | Encourage team-specific or sensitive projects that require restricted access. |
+| ┗ Internal | Members can create internal repositories visible to organization members. | Foster collaboration and knowledge sharing within the organization while keeping information proprietary. |
+
+#### Repository forking (Private / Internal)
+
+The decision to allow or restrict repository forking depends on your organization's policies and the level of control and security required for your codebase. Enabling forking generally promotes collaboration, allows for experimentation, and encourages the InnerSource principles of sharing and contribution. However, it's important to assess the risks and consider any regulatory or compliance requirements before allowing forking in private and internal repositories.
```

Comment:
The risk that people in the company are thinking about would be being cloned into a private repository, but honestly that can be easily prevented by configuration. Also, with a fork in the enterprise, if the original was erased, the fork would have disappeared as well.
I think we should be clear about what the risk is.

Basically, there shouldn't be that big of a risk, but the enterprise that wants to control it may not be pleased to have more forks that are not maintained out of their control.
In that sense, I think it would be better to point out the recommendations for setting up the enterprise, mentioning not the risk but the operational resistance.

What's your environment like here? If you could share...